src/main/java/org/jboss/remoting3/EndpointBuilder.java

```diff
@@ -40,16 +42,63 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class EndpointBuilder {
+
+    private static final Logger log = Logger.getLogger("org.jboss.remoting.endpoint");
+
     private String endpointName;
     private XnioWorker xnioWorker;
     private List<ConnectionProviderFactoryBuilder> connectionProviderFactoryBuilders;
     private List<ConnectionBuilder> connectionBuilders;
     private XnioWorker.Builder workerBuilder;
-    //Default option map that sets heartbeat and read/write timeouts
-    private OptionMap defaultConnectionOptionMap = OptionMap.builder().set(RemotingOptions.HEARTBEAT_INTERVAL, RemotingOptions.DEFAULT_HEARTBEAT_INTERVAL)
+
+    private static final String MAX_INBOUND_CHANNELS_KEY = "remoting.endpoint.max.inbound.channels";
+    private static final String MAX_OUTBOUND_CHANNELS_KEY = "remoting.endpoint.max.outbound.channels";
+    
+    private static final int MAX_INBOUND_CHANNELS;
+    private static final int MAX_OUTBOUND_CHANNELS;
+
+    static {
```

Comment:
This static initializer should be in RemotingOptions class IMO and it should change the default values of
RemotingOptions.DEFAULT_MAX_INBOUND_CHANNELS and
RemotingOptions.DEFAULT_MAX_OUTBOUND_CHANNELS
because these defaults are used in both ClientConnectionOpenListener & ServerConnectionOpenListener classes.